pressbooks/pressbooks
# feat: add Algolia integration for clone page

Related issue: https://github.com/pressbooks/pressbooks/issues/2317

This PR integrates Instantsearch and Algolia libraries with a basic HTML structure to search, display and pick from the books list in our [Algolia's indexes used in Pressbooks Directory](https://github.com/pressbooks/pressbooks-book-directory-fe) and complete the `Book URL` field in the cloning page with a single click of a button.

To enable the feature, 3 ENV variables must be defined:
```
ALGOLIA_APP_ID=algolia_api_id
ALGOLIA_API_KEY=algolia_api_key
ALGOLIA_INDEX_NAME=algolia_index_name
```
You must get these values from the Algolia app.

Run `lando npm install` to install new dependencies.

### Test case
- Go to the clone page.
- You should see a new search input at the bottom.
- Type in the search box, and a list of books from the Directory should be listed at the bottom of the page in the form of book cards.
- Each book card contains a "Select" button. If you click there, the Book URL must be autocompleted with the selected Book URL.
